Add ASA Attributes (#1513)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/asa/attributes/index.md

This endpoint lists the Attributes for an ASA User.

This endpoint requires one of the following roles: `access_admin`, `access_user`, `reporting_user`.

#### Request path parameters

| Parameter | Type        | Description   |
| --------- | ----------- | ------------- |
| `team_name`   | string | The name of your Team. |
| `user_name`   | string | The relevant username. |

#### Request query parameters

| Parameter | Type   | Description |
| --------- | ------------- | -------- |
| `conflicting`   |  boolean | (Optional) When true, returns only attributes that conflict with other ASA User attributes on this Team. |
| `count`   |  number | (Optional) The number of objects per page. |
| `descending`   |  boolean | (Optional) The object order. |
| `offset`   |  string | (Optional) The page offset. |
| `prev`   |  boolean | (Optional) The direction of paging. |

#### Request body

This endpoint has no request body.

#### Response body
This endpoint returns a list of objects with the following fields and a `200` code on a successful call.
| Properties | Type        | Description          |
|----------|-------------|----------------------|

> | `attribute_name`   | string | One of: `unix_user_name`, `unix_uid`, `unix_gid`, `windows_user_name`. |

Reviewer: ```suggestion
| `attribute_name`   | string | Accepted values: `unix_user_name`, `unix_gid`, `unix_uid`, or `windows_user_name` |
```

Please do this for all `attribute_name` table entries.

> | `attribute_value`   | object | The value of the attribute. Must be a number for uid/gid, a string for any other attribute type. |

Reviewer: ```suggestion
| `attribute_value`   | object | Accepted values for `unix_user_name` and `windows_user_name` are strings with a character length between 0 and 255. Accepted values for `unix_gid` and `unix_uid` are numbers between 100 and 2147483647. |
```
Please do this for all `attribute_value` table entries.

| `id`   | string | The unique identifier for the attribute. |
| `managed`   | boolean | Whether or not this attribute is being managed through SCIM or SAML. |

#### Usage example

##### Request

```bash
curl -v -X GET \
-H "Authorization: Bearer ${jwt}" \
https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}/attributes
```

##### Response

```json
{
	"list": [
		{
			"attribute_name": "unix_user_name",
			"attribute_value": "augusta_ada_king",
			"id": "11faefa1-6b59-4a52-9492-43195cd07385",
			"managed": true
		},
		{
			"attribute_name": "unix_uid",
			"attribute_value": 1210,
			"id": "3a95c65d-d1dc-4642-b25b-5b74d5bd8da5",
			"managed": true
		},
		{
			"attribute_name": "unix_gid",
			"attribute_value": 1210,
			"id": "b06dfc59-ae9c-4243-b583-96d09988fd84",
			"managed": true
		},
		{
			"attribute_name": "windows_user_name",
			"attribute_value": "augusta_ada_king",
			"id": "ff2c0f25-b73d-4720-8aeb-b6c39a68a204",
			"managed": true
		}
	]
}
```
### Fetch a User Attribute

<ApiOperation method="GET" url="https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}/attributes/${attribute_id}" />

> This endpoint fetches the details of an Attribute for an ASA User.

Reviewer: ```suggestion
Fetches the details of an Attribute for an ASA User
```

> This endpoint requires one of the following roles: `access_admin`, `access_user`, `reporting_user`.

Reviewer: ```suggestion
This endpoint requires one of the following roles: `access_admin`, `access_user`, or `reporting_user`.
```

This endpoint has no request body.

#### Response body
This endpoint returns an object with the following fields and a `200` code on a successful call.
| Properties | Type        | Description          |
|----------|-------------|----------------------|
| `attribute_name`   | string | One of: `unix_user_name`, `unix_uid`, `unix_gid`, `windows_user_name`. |
| `attribute_value`   | object | The value of the attribute. Must be a number for uid/gid, a string for any other attribute type. |
| `id`   | string | The unique identifier for the attribute. |
| `managed`   | boolean | Whether or not this attribute is being managed through SCIM or SAML. |

#### Usage example

##### Request

```bash
curl -v -X GET \
-H "Authorization: Bearer ${jwt}" \
https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}/attributes/${attribute_id}
```

##### Response

```json
{
	"attribute_name": "unix_user_name",
	"attribute_value": "augusta_ada_king",
	"id": "11faefa1-6b59-4a52-9492-43195cd07385",
	"managed": true
}
```
### Update a single Attribute for a User

<ApiOperation method="PUT" url="https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}/attributes/${attribute_id}" />

> This endpoint updates an Attribute for an ASA User.

Reviewer: ```suggestion
Updates an Attribute for an ASA User
```

> This endpoint requires one of the following roles: `access_admin`.

Reviewer: ```suggestion
This endpoint requires the `access_admin` role.
```
